packages/query/src/__tests__/query.test.ts

```diff
@@ -176,6 +176,42 @@ describe('query', () => {
     })
   })
 
+  it('limit and sorting', async (done) => {
+    const storage = await getClient()
+
+    const limit = 1
+    let attempt = 0
+    let doneCount = 0
+
+    const query = withOperations(core.account.System, new LiveQuery(storage))
+    query.query<Space>(core.class.Space, { private: true }, (result) => {
```

Comment:
Could `new Promise((resolve)=> ...) `make it a bit more clear, so we could not use doneCount in this case.